iroh-base/src/key.rs

```diff
@@ -384,14 +380,14 @@ fn decode_base32_hex(s: &str) -> Result<[u8; 32], KeyParsingError> {
         let input = s.to_ascii_uppercase();
         let input = input.as_bytes();
         if data_encoding::BASE32_NOPAD.decode_len(input.len())? != bytes.len() {
-            return Err(DecodeInvalidLengthSnafu.build());
+            return Err!(KeyParsingError::DecodeInvalidLength);
         }
         data_encoding::BASE32_NOPAD.decode_mut(input, &mut bytes)
     };
     match res {
         Ok(len) => {
             if len != PublicKey::LENGTH {
-                return Err(DecodeInvalidLengthSnafu.build());
+                return Err!(KeyParsingError::DecodeInvalidLength);
```

Comment:
looking at this, I would much rather write

```rust
bail!(KeyParsingError::DecodeInvalidLength);
```

overloading `Err`  I find problematic personally

Reply:
I think that would be better to have than `Err!`